tools/update_extern.py

```diff
@@ -0,0 +1,169 @@
+#!/bin/env python
+
+"""
+This script is used to update the files in `sunpy/extern` while making new releases.
+"""
+
+import os
+import shutil
+from zipfile import ZipFile
+import requests
+from pathlib import Path
+
+# get parent directory of sunpy
+SUNPY_DIR = Path(__file__).parent.parent
+
+
+# List of packages to update
+PACKAGES = ["appdirs", "distro", "inflect", "parse"]
+
+# Dictionary to store the authors of the packages
+AUTHORS = {
+    "appdirs": "ActiveState",
+    "distro": "python-distro",
+    "inflect": "jaraco",
+    "parse": "r1chardj0n3s",
+}
+
+
+def get_latest_version(package):
+    """
+    Get the latest version of the package.
+    """
+    url = f"https://api.github.com/repos/{AUTHORS[package]}/{package}/releases/latest"
+    response = requests.get(url)
+    return response.json()["tag_name"]
+
+
+def get_download_url(package, version):
+    """
+    Get the download url for the package.
+    """
+    url = f"https://github.com/{AUTHORS[package]}/{package}/archive/refs/tags/{version}.zip"
+    return url
+
+
+def download_package(url, package):
+    """
+    Download the package.
+    """
+    # Make a temporaty directory
+    if not os.path.exists("extern_pkg"):
+        os.mkdir("extern_pkg")
+
+    response = requests.get(url, stream=True)
+    with open(f"extern_pkg/{package}.zip", "wb") as f:
+        print(f"Downloading {package}")
+        for chunk in response.iter_content(chunk_size=8192):
+            if chunk:
+                f.write(chunk)
+            f.flush()
+        f.close()
+
+
+def unzip(filename):
+    """
+    Extract the package.
+    """
+    with ZipFile(filename, "r") as zip_file:
+        zip_file.extractall("extern_pkg")
+    # Remove the zip file
+    os.remove(filename)
+
+
+def move_files(src: Path, dst: Path):
+    """
+    Move the files from the src to the dst.
+    """
+    # if dst already exists, remove it
+    if os.path.exists(dst):
+        print(f"Removing {dst}")
+    # move the files
+    shutil.move(src, dst)
+    print(f"Moving {src} to {dst}")
+
+
+def extract_appdirs(destination):
```

Comment:
Do we need one function per library?

Reply:
Earlier I thought the same thing. But the problem is, that the file which needs to be extracted from the source distribution is not the same for every package. 
For example, in `appdirs`, we need `appdirs/appdirs.py` but in `inflect`, we need `inflect/inflect/__init__.py`.
for `distro` we need `src/distro/distro.py`, for `parse` we need `/parse.py`. 
I thought that having a different function to extract the required `.py` for each package will be better than having a single complicated function.

If there is any better way we can work out, I will do it.